st2actions/st2actions/container/base.py

```diff
@@ -186,6 +186,10 @@ def _do_cancel(self, runner, runnertype_db, action_db, liveaction_db):
                 liveaction_db=liveaction_db)
 
             executions.update_execution(liveaction_db)
+
+            LOG.debug('Performing post_run for runner: %s', runner.runner_id)
```

Comment:
Hm, do we also want to execute this piece of code on cancel - https://github.com/StackStorm/st2/blob/2da0bfc42f5fb36453b1cd1c463ff49cd45942c7/st2actions/st2actions/container/base.py#L154?

It seems like we do (deleting the auth token)

Current approach looks fine to me for now, but it looks like some refactoring is in order.